### DISCUSSION TITLE
Preparation for gdUnit4?

### DISCUSSION THREAD

**Discussion by GeorgeS2019** [MikeSchulze/gdUnit3#226] [Ideas]
Currently the master Godot4 mono is working. The Godot4 dotnet6 branch can be compiled ([at least in Windows](https://gist.github.com/lewiji/2b86f47dcb142a10e104504a343c38ff)) and is making a good progress. 

User can now work with Godot4 mono and create project. It is likely with minimum effort that the same project and code in c# and GDscript can be ported to Godot4 dotnet6 after merged.

Since there is a vision of gdUnit4 (personal communication), this discussion is to track what need to be prepared and planned to transition from gdUnit3 to gdUnit4. 

Perhaps gdUnit4 can be used by community to assist in getting Godot4 .NET6 mature and robust. :-)

**Comment by GeorgeS2019** [comment]
The [link ](https://github.com/godotengine/godot-proposals/discussions/4416#dotnet6-test)to the godot4 discussion for supporting the transition from mono to dotnet6

**Reply by MikeSchulze** [reply]
GdUnit3 is using .Net6 since version 2.2.0 (next coming release)

**Comment by GeorgeS2019** [comment]
No pressure,  Godot4 dotnet6 merged!  Again no pressure. U have done an excellent job!

**Comment by MikeSchulze** [comment]
Migration to GdUnit4 has started
https://github.com/MikeSchulze/gdUnit4
